### ISSUE TITLE
Improve interfaces between cli.subcmd.init & DBConnector

### ISSUE THREAD

**Issue by marcus-grant** [marcus-grant/scout#24]
The subcommand 'init' to the CLI is very cumbersome and
repeats a lot of work the `DBConnector` class already does or
should be doing.
Namely all the exception handling should be done by
the `DBConnector` class and not by the CLI.
Then a **try-catch** block in the CLI subcommand can be used to
determine if the database initialization worked.
If it doesn't the `DBConnector` should have its own Error types to
make it clear to other modules what went wrong.
Then the CLI can give the helpful and more user friendly messages.

The `init` subcommand should also have an alternative way to
initialize the database that
doesn't involve constructing the `ScoutManager`.
It should call the `DBConnector.init_db` method directly.
Since the command just initializes and exits with a message,
there's really no point in involving the `ScoutManager`.
Or maybe there is I'm not sure,
it would be nice if nothing interacts with repo classes directly.
If you do involve `ScoutManager` it should probably be a class function that
forward the call to the `DBConnector` class.

TLDR;
- [x] Improve the `DBConnector` exception handling with own error types.
    - Handled in 2b34616
- [x] Make the `init` subcommand more friendly, by responding to those errors.
- [x] Consider if the `ScoutManager` should be involved in the `init` subcommand.
- [x] Have `cli.subcmd.init.handle_subcommand`...`
  - [x] Call `DBConnector.init_db` or `ScoutManager.init_db` directly.
  - [x] Handle exceptions with user friendly messages and return codes.
  - [x] Catch the new `DBConnector` error types.
- [x] Link this issue to the `init` subcommand issue #21

**Comment by marcus-grant** [comment]
First checklist almost completed with 1fc1212 , continue by checking `DBConnector` `TODO` comments.

**Comment by marcus-grant** [comment]
First checklist item handled in 2b34616

**Comment by marcus-grant** [comment]
Continuing now that issue #26 resolved in 362e8b4

**Comment by marcus-grant** [comment]
Handled in PR #28 
The other checkmarks were either handled in seprate issues or were dupes.
